modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/clip/ShareClipFragment.kt

```diff
@@ -81,22 +89,38 @@ class ShareClipFragment : BaseDialogFragment() {
         savedInstanceState: Bundle?,
     ) = ComposeView(requireActivity()).apply {
         val platforms = SocialPlatform.getAvailablePlatforms(requireContext())
-        val assetController = BackgroundAssetController.create(requireContext(), shareColors)
-        val listener = shareListenerFactory.create(this@ShareClipFragment, viewModel, assetController, args.source)
+        val assetController = BackgroundAssetController.create(requireContext().applicationContext, shareColors)
+        val listener = ShareClipListener(this@ShareClipFragment, viewModel, assetController, args.source)
+
         setContent {
             val uiState by viewModel.uiState.collectAsState()
+            val snackbarHostState = remember { SnackbarHostState() }
+
             ShareClipPage(
                 episode = uiState.episode,
                 podcast = uiState.podcast,
                 clipRange = uiState.clipRange,
                 playbackProgress = uiState.playbackProgress,
                 isPlaying = uiState.isPlaying,
+                isSharing = uiState.isSharing,
                 useEpisodeArtwork = uiState.useEpisodeArtwork,
                 platforms = platforms,
                 shareColors = shareColors,
                 assetController = assetController,
                 listener = listener,
+                snackbarHostState = snackbarHostState,
             )
+
+            LaunchedEffect(Unit) {
+                viewModel.snackbarMessages.collect { message ->
+                    val text = when (message) {
+                        is SnackbarMessage.SharingResponse -> message.message
+                        is SnackbarMessage.PlayerIssue -> getString(LR.string.podcast_episode_playback_error)
+                        is SnackbarMessage.GenericIssue -> getString(LR.string.error)
```

Comment:
Maybe we should have a nicer error message here, something like "Unable to create clip. Please try again later."

![Screenshot_20240809_135748_Pocket Debug](https://github.com/user-attachments/assets/59701f1e-a523-4128-91e3-0b90f93665ca)

Reply:
Sure, makes sense 👍

Reply:
Fixed in b41a92b58da8bbdb961abce002874a7d411d38b8